### PR TITLE
[compiler] IndexWriter: don't flush after every record

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Review by danking — CHANGES_REQUESTED** [review]
We appear to have two load-bearing flushes.

Hmm. Do we ever close the OutputBuffer? That seems like a bug.

File: hail/src/main/scala/is/hail/io/index/IndexWriter.scala [hunk 1]
```diff
@@ -370,6 +370,8 @@ class StagedIndexWriter(branchingFactor: Int, keyType: PType, annotationType: PT
   }
   def close(cb: EmitCodeBuilder): Unit = {
     val off = flush.invokeCode[Long](cb)
+    cb += ob.flush()
+    cb += ob.close()
```
Comment (danking) [inline hail/src/main/scala/is/hail/io/index/IndexWriter.scala:374]: I think we lost the removal of the other flushes in writeLeafNode and writeInternalNode.

**Review by danking — CHANGES_REQUESTED** [review]

File: hail/src/main/scala/is/hail/io/index/IndexWriter.scala (hunk 1 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/io/index/IndexWriter.scala:374]: 🤦 

no wonder it worked...

**Comment by tpoterba** [comment]
understand the issue -- need to fix the buffer spec used by the index writer.

**Pull request closed without merging**